handler_context/handler_context.go

```diff
@@ -0,0 +1,60 @@
+package handler_context
+
+import "errors"
+
+type HandlerContext interface {
+	Skip(runnable) bool
+	MarkAsDone(runnable) error
+	Stop(string)
+	Stopped() bool
+	StopReason() string
+}
+
+type runnable interface {
+	Name() string
+}
+
+type handlerContext struct {
+	doneStages map[string]bool
+	stopped bool
+	stopReason string
+}
+
+func CreateFrom(previousCtx HandlerContext) *handlerContext {
+	if previousCtx == nil {
+		return &handlerContext{}
+	}
+
+	if castedCtx, ok := previousCtx.(*handlerContext); ok {
+		return castedCtx
+	}
+
+	return &handlerContext{}
+}
+
+func (ctx *handlerContext) Skip(stage runnable) bool {
```

Comment:
It sounds like a command where the code (if i understand it correctly) is a test is this phase was mark as done.